addons/delivery/wizard/choose_delivery_package.py

```diff
@@ -8,42 +8,19 @@ class ChooseDeliveryPackage(models.TransientModel):
     _name = 'choose.delivery.package'
     _description = 'Delivery Package Selection Wizard'
 
-    stock_quant_package_id = fields.Many2one(
-        'stock.quant.package',
-        string="Physical Package",
-        default=lambda self: self._default_stock_quant_package_id()
-    ) 
-    delivery_packaging_id = fields.Many2one(
-        'product.packaging',
-        default=lambda self: self._default_delivery_packaging_id()
-    )
-    shipping_weight = fields.Float(
-        string='Shipping Weight',
-        default=lambda self: self._default_shipping_weight()
-    )
-
-    def _default_stock_quant_package_id(self):
-        if self.env.context.get('default_stock_quant_package_id'):
-            return self.env['stock.quant.package'].browse(self.env.context['stock_quant_package_id'])
-
-    def _default_delivery_packaging_id(self):
-        res = None
-        if self.env.context.get('default_delivery_packaging_id'):
-            res = self.env['product.packaging'].browse(self.env.context['default_delivery_packaging_id'])
-        if self.env.context.get('default_stock_quant_package_id'):
-            stock_quant_package = self.env['stock.quant.package'].browse(self.env.context['default_stock_quant_package_id'])
-            res = stock_quant_package.packaging_id
-        return res
+    picking_id = fields.Many2one('stock.picking', 'Picking')
+    delivery_packaging_id = fields.Many2one('product.packaging', 'Delivery Packaging')
+    shipping_weight = fields.Float('Shipping Weight', default=lambda self: self._default_shipping_weight())
 
     def _default_shipping_weight(self):
-        if self.env.context.get('default_stock_quant_package_id'):
-            stock_quant_package = self.env['stock.quant.package'].browse(self.env.context['default_stock_quant_package_id'])
-            return stock_quant_package.shipping_weight
-        else:
-            picking_id = self.env['stock.picking'].browse(self.env.context['active_id'])
-            move_line_ids = [po for po in picking_id.move_line_ids if po.qty_done > 0 and not po.result_package_id]
-            total_weight = sum([po.qty_done * po.product_id.weight for po in move_line_ids])
-            return total_weight
+        pid = self.env.context.get('default_picking_id')
+        picking_id = self.env['stock.picking'].browse(pid)
+        move_line_ids = [ml for ml in picking_id.move_line_ids if ml.qty_done > 0 and not ml.result_package_id]
```

Comment:
picking_id.move_line_ids.filtered(lambda ml: ml.qty_done > 0 and not ml.result_package_id)